from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(567, 237)
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.tableView = QtWidgets.QTableView(self.centralwidget)
        self.tableView.setGeometry(QtCore.QRect(10, 10, 341, 181))
        font = QtGui.QFont()
        font.setPointSize(12)
        self.tableView.setFont(font)
        self.tableView.setObjectName("tableView")
        self.frame = QtWidgets.QFrame(self.centralwidget)
        self.frame.setGeometry(QtCore.QRect(350, 10, 201, 181))
        self.frame.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.frame.setFrameShadow(QtWidgets.QFrame.Raised)
        self.frame.setObjectName("frame")
        self.gridLayout = QtWidgets.QGridLayout(self.frame)
        self.gridLayout.setObjectName("gridLayout")
        self.logfileButton = QtWidgets.QPushButton(self.frame)
        self.logfileButton.setObjectName("logfileButton")
        self.gridLayout.addWidget(self.logfileButton, 6, 0, 1, 1)
        self.refreshRateButton = QtWidgets.QPushButton(self.frame)
        self.refreshRateButton.setObjectName("refreshRateButton")
        self.gridLayout.addWidget(self.refreshRateButton, 4, 0, 1, 1)
        self.resetMinMaxButton = QtWidgets.QPushButton(self.frame)
        self.resetMinMaxButton.setObjectName("resetMinMaxButton")
        self.gridLayout.addWidget(self.resetMinMaxButton, 5, 0, 1, 1)
        self.refreshLabel = QtWidgets.QLabel(self.frame)
        self.refreshLabel.setAlignment(QtCore.Qt.AlignLeading|QtCore.Qt.AlignLeft|QtCore.Qt.AlignVCenter)
        self.refreshLabel.setObjectName("refreshLabel")
        self.gridLayout.addWidget(self.refreshLabel, 2, 0, 1, 1)
        self.nameLabel = QtWidgets.QLabel(self.frame)
        self.nameLabel.setObjectName("nameLabel")
        self.gridLayout.addWidget(self.nameLabel, 0, 0, 1, 1)
        self.tempLabel = QtWidgets.QLabel(self.frame)
        self.tempLabel.setAlignment(QtCore.Qt.AlignLeading|QtCore.Qt.AlignLeft|QtCore.Qt.AlignVCenter)
        self.tempLabel.setObjectName("tempLabel")
        self.gridLayout.addWidget(self.tempLabel, 1, 0, 1, 1)
        MainWindow.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(MainWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 567, 22))
        self.menubar.setObjectName("menubar")
        self.menuFile = QtWidgets.QMenu(self.menubar)
        self.menuFile.setObjectName("menuFile")
        self.menuSettings = QtWidgets.QMenu(self.menubar)
        self.menuSettings.setObjectName("menuSettings")
        self.menuHelp = QtWidgets.QMenu(self.menubar)
        self.menuHelp.setObjectName("menuHelp")
        self.menuDevices = QtWidgets.QMenu(self.menubar)
        self.menuDevices.setObjectName("menuDevices")
        MainWindow.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(MainWindow)
        self.statusbar.setObjectName("statusbar")
        MainWindow.setStatusBar(self.statusbar)
        self.actionClose = QtWidgets.QAction(MainWindow)
        self.actionClose.setObjectName("actionClose")
        self.actionOpen_New_Window = QtWidgets.QAction(MainWindow)
        self.actionOpen_New_Window.setShortcut("")
        self.actionOpen_New_Window.setObjectName("actionOpen_New_Window")
        self.actionReset_Min_Max = QtWidgets.QAction(MainWindow)
        self.actionReset_Min_Max.setObjectName("actionReset_Min_Max")
        self.actionSet_Refresh_Rate = QtWidgets.QAction(MainWindow)
        self.actionSet_Refresh_Rate.setObjectName("actionSet_Refresh_Rate")
        self.actionView_Available_devices = QtWidgets.QAction(MainWindow)
        self.actionView_Available_devices.setObjectName("actionView_Available_devices")
        self.actionAbout = QtWidgets.QAction(MainWindow)
        self.actionAbout.setObjectName("actionAbout")
        self.actionCreateCSV_log_file = QtWidgets.QAction(MainWindow)
        self.actionCreateCSV_log_file.setObjectName("actionCreateCSV_log_file")
        self.menuFile.addAction(self.actionOpen_New_Window)
        self.menuFile.addAction(self.actionClose)
        self.menuSettings.addAction(self.actionReset_Min_Max)
        self.menuSettings.addAction(self.actionSet_Refresh_Rate)
        self.menuSettings.addAction(self.actionCreateCSV_log_file)
        self.menuHelp.addAction(self.actionAbout)
        self.menubar.addAction(self.menuFile.menuAction())
        self.menubar.addAction(self.menuDevices.menuAction())
        self.menubar.addAction(self.menuSettings.menuAction())
        self.menubar.addAction(self.menuHelp.menuAction())

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "MainWindow"))
        self.logfileButton.setText(_translate("MainWindow", "Create log file"))
        self.refreshRateButton.setText(_translate("MainWindow", "Change Refresh Rate"))
        self.resetMinMaxButton.setText(_translate("MainWindow", "Reset Min/Max"))
        self.refreshLabel.setText(_translate("MainWindow", "Refresh Rate: 10s"))
        self.nameLabel.setText(_translate("MainWindow", "Temperature Monitor v1.0"))
        self.tempLabel.setText(_translate("MainWindow", "Temperate in Celsius"))
        self.menuFile.setTitle(_translate("MainWindow", "File"))
        self.menuSettings.setTitle(_translate("MainWindow", "Tools"))
        self.menuHelp.setTitle(_translate("MainWindow", "Help"))
        self.menuDevices.setTitle(_translate("MainWindow", "Devices"))
        self.actionClose.setText(_translate("MainWindow", "Close"))
        self.actionClose.setShortcut(_translate("MainWindow", "Ctrl+C"))
        self.actionOpen_New_Window.setText(_translate("MainWindow", "Open New Window"))
        self.actionReset_Min_Max.setText(_translate("MainWindow", "Reset Min/Max"))
        self.actionSet_Refresh_Rate.setText(_translate("MainWindow", "Set Refresh Rate"))
        self.actionView_Available_devices.setText(_translate("MainWindow", "View Available devices"))
        self.actionAbout.setText(_translate("MainWindow", "About"))
        self.actionAbout.setShortcut(_translate("MainWindow", "Ctrl+H"))
        self.actionCreateCSV_log_file.setText(_translate("MainWindow", "CreateCSV log file"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    MainWindow = QtWidgets.QMainWindow()
    ui = Ui_MainWindow()
    ui.setupUi(MainWindow)
    MainWindow.show()
    sys.exit(app.exec_())
